File: create_tilda_landing.py
#!/usr/bin/env python3
"""
Скрипт для создания 5-блочного лендинга на Tilda
Курс: "Магия женского мира: ведьмин клубок и инициация"
Папка: Светлана Комарова
URL: magic
"""

import asyncio
import os
import sys
import time
from pathlib import Path

try:
    from playwright.async_api import async_playwright, TimeoutError as PlaywrightTimeoutError
except ImportError:
    print("Playwright не установлен. Устанавливаем...")
    os.system("pip install playwright")
    os.system("playwright install chromium")
    from playwright.async_api import async_playwright, TimeoutError as PlaywrightTimeoutError

# Конфигурация
TILDA_LOGIN = "Benedictuminfo@yandex.ru"
TILDA_PASSWORD = "HisJY25x6x&iZq4"
TARGET_FOLDER = "Светлана Комарова"
PAGE_URL = "magic"
IMAGES_DIR = Path(__file__).parent / "extracted_images" / "word" / "media"

# Содержимое лендинга
LANDING_CONTENT = {
    "title": "Магия женского мира: ведьмин клубок и инициация",
    "subtitle": "Двухнедельное погружение в алгоритмы женской инициации через призму русских сказок",
    "start_date": "Старт 23 марта",
    "about": """Женская инициация в сказках — обязательный этап, превращающий девушку в женщину, способную управлять своей жизнью.

Переход происходит по трёхчастной модели: отделение → трансформация → возвращение.

Остановка на любом этапе создаёт замороженный сценарий.""",
    "symptoms": [
        "Точка изгнания — потеря привычных социальных ролей воспринимается как катастрофа",
        "Застрявшая в «тёмном лесу» — неспособность распознать проводников и важные знаки",
        "Потеря нити — отсутствие внутреннего компаса (клубок и зеркало для правды)",
        "Отказ от короны — страх взять ответственность Принцессы или мудрость Хранительницы",
    ],
    "week1": [
        {"date": "23 марта", "title": "Модель инициации", "desc": "Три стадии: отделение, трансформация, возвращение"},
        {"date": "25 марта", "title": "Архетипы и символы", "desc": "Роль Бабы-яги и сакральные инструменты: клубок, зеркало, прялка, ключ, вода"},
        {"date": "27 марта", "title": "Итоги перехода", "desc": "Маркеры зрелости и критерии нового статуса"},
    ],
    "week2": [
        {"date": "31 марта", "title": "Повторение как проверка", "desc": "Механизм подтверждения зрелости через паттерны"},
        {"date": "2 апреля", "title": "Законы обмена", "desc": "Личное счастье через служение и баланс «брать» и «давать»"},
        {"date": "3 апреля", "title": "Спецэфир", "desc": "Только для Внутреннего и Частного кругов. Александра Демьяненко — ведьмина инициация или бытовая магия"},
    ],
    "speakers": [
        {
            "name": "Светлана Комарова",
            "role": "Бизнес-психолог, автор книги «Бизнес ведьмы», основатель Академии Benedictum",
            "image": str(IMAGES_DIR / "image1.png"),
        },
        {
            "name": "Елена Соломина",
            "role": "Член ОППЛ, специалист по глубинной групповой психологии, эксперт по влиянию сказок на жизнь",
            "image": str(IMAGES_DIR / "image3.png"),
        },
        {
            "name": "Александра Демьяненко",
            "role": "Эксперт по русскому фольклору и мифологии, психолог",
            "image": str(IMAGES_DIR / "image4.png"),
        },
    ],
    "tariffs": [
        {
            "name": "Внешний круг",
            "price": "14 700 ₽",
            "early_price": "13 300 ₽",
            "features": ["Доступ к вебинарам как слушатель"],
        },
        {
            "name": "Внутренний круг",
            "price": "21 100 ₽",
            "early_price": "18 800 ₽",
            "features": [
                "Интерактивная работа с обратной связью",
                "Онлайн-встречи",
                "Эксклюзивный эфир Демьяненко о бытовой магии",
            ],
        },
        {
            "name": "Частный круг",
            "price": "33 300 ₽",
            "early_price": "30 300 ₽",
            "features": [
                "Максимальная плотность работы",
                "Индивидуальный разбор",
                "Всё из Внутреннего круга",
                "Закрытая групповая консультация со спикерами (до 10 человек)",
            ],
        },
    ],
}


async def wait_and_click(page, selector, timeout=15000):
    """Ждёт элемент и кликает по нему"""
    await page.wait_for_selector(selector, timeout=timeout)
    await page.click(selector)


async def login_to_tilda(page):
    """Вход в Tilda"""
    print("Открываем страницу входа...")
    await page.goto("https://tilda.cc/login/", wait_until="networkidle")
    await page.wait_for_timeout(2000)

    print("Вводим логин и пароль...")
    # Ищем поля ввода
    await page.fill('input[name="email"], input[type="email"], #email', TILDA_LOGIN)
    await page.fill('input[name="password"], input[type="password"], #password', TILDA_PASSWORD)

    # Кликаем кнопку входа
    await page.click('button[type="submit"], input[type="submit"], .js-form-submit, [data-form-submit]')

    await page.wait_for_load_state("networkidle")
    await page.wait_for_timeout(3000)

    current_url = page.url
    print(f"После входа URL: {current_url}")

    if "login" in current_url:
        raise Exception("Вход не выполнен — возможно неверные данные")

    print("Вход выполнен успешно!")


async def find_svetlana_folder(page):
    """Находим папку Светлана Комарова и получаем project_id"""
    print("Ищем папку 'Светлана Комарова'...")

    # Переходим на главную страницу проектов
    await page.goto("https://tilda.cc/projects/", wait_until="networkidle")
    await page.wait_for_timeout(3000)

    # Ищем папку по тексту
    folder_link = page.locator(f'text="{TARGET_FOLDER}"').first
    if not await folder_link.is_visible():
        # Пробуем другие варианты поиска
        folder_link = page.locator(f':text("{TARGET_FOLDER}")').first

    if await folder_link.is_visible():
        print(f"Папка '{TARGET_FOLDER}' найдена, кликаем...")
        await folder_link.click()
        await page.wait_for_load_state("networkidle")
        await page.wait_for_timeout(2000)
        project_url = page.url
        print(f"URL проекта: {project_url}")
        return project_url
    else:
        # Делаем скриншот для диагностики
        await page.screenshot(path="/tmp/tilda_projects.png")
        print("Сохранён скриншот: /tmp/tilda_projects.png")
        raise Exception(f"Папка '{TARGET_FOLDER}' не найдена")


async def get_font_from_existing_page(page, project_url):
    """Получаем шрифт из существующих лендингов (не из 'Другая')"""
    print("Определяем шрифт из существующих лендингов...")

    # Переходим обратно к проектам и ищем другой проект (не Другая) для получения шрифта
    await page.goto("https://tilda.cc/projects/", wait_until="networkidle")
    await page.wait_for_timeout(2000)

    # Ищем все проекты кроме "Другая"
    projects = await page.query_selector_all(".projects-list__item, .project-card, [data-project]")

    font_name = None
    for project in projects:
        project_text = await project.inner_text()
        if "Другая" not in project_text and TARGET_FOLDER in project_text:
            # Это наш проект, попробуем найти шрифт в его настройках
            pass

    # Возвращаем шрифт по умолчанию для аккаунта (Histori Pro указан в документе)
    # Но Роман сказал брать с других лендов - попробуем через API
    return font_name


async def find_project_id_from_url(page):
    """Извлекаем ID проекта из URL"""
    current_url = page.url
    # URL вида: https://tilda.cc/projects/?projectid=12345
    if "projectid=" in current_url:
        return current_url.split("projectid=")[1].split("&")[0]
    return None


async def create_new_page_via_ui(page, project_url):
    """Создаём новую страницу через UI Tilda"""
    print("Создаём новую страницу...")

    # Переходим в проект
    await page.goto(project_url, wait_until="networkidle")
    await page.wait_for_timeout(3000)

    await page.screenshot(path="/tmp/tilda_project.png")
    print("Скриншот проекта: /tmp/tilda_project.png")

    # Ищем кнопку создания страницы
    add_page_selectors = [
        "text=Добавить страницу",
        "text=Add page",
        ".js-add-page",
        "[data-add-page]",
        ".addpage-btn",
        "button:has-text('+')",
    ]

    clicked = False
    for selector in add_page_selectors:
        try:
            if await page.is_visible(selector, timeout=3000):
                await page.click(selector)
                clicked = True
                break
        except Exception:
            continue

    if not clicked:
        await page.screenshot(path="/tmp/tilda_no_add_btn.png")
        raise Exception("Не нашли кнопку добавления страницы")

    await page.wait_for_timeout(3000)
    await page.screenshot(path="/tmp/tilda_add_page_dialog.png")

    return True


async def configure_page_settings(page):
    """Настройка параметров новой страницы"""
    print("Настраиваем параметры страницы...")

    # Ищем поле для имени страницы
    name_selectors = [
        'input[name="title"]',
        'input[placeholder*="название"]',
        'input[placeholder*="title"]',
        '#page-title',
    ]

    for selector in name_selectors:
        try:
            if await page.is_visible(selector, timeout=3000):
                await page.fill(selector, LANDING_CONTENT["title"])
                break
        except Exception:
            continue

    # Ищем поле URL
    url_selectors = [
        'input[name="alias"]',
        'input[name="url"]',
        'input[placeholder*="url"]',
        'input[placeholder*="URL"]',
        '#page-alias',
    ]

    for selector in url_selectors:
        try:
            if await page.is_visible(selector, timeout=3000):
                await page.fill(selector, PAGE_URL)
                break
        except Exception:
            continue

    await page.wait_for_timeout(1000)


async def use_tilda_api(page):
    """Используем Tilda API для получения информации о проекте"""
    print("Получаем данные через Tilda API...")

    # Сначала нужно получить API ключи из настроек аккаунта
    await page.goto("https://tilda.cc/account/", wait_until="networkidle")
    await page.wait_for_timeout(2000)

    await page.screenshot(path="/tmp/tilda_account.png")

    # Ищем API ключи
    api_content = await page.content()

    public_key = None
    secret_key = None

    # Пробуем найти API ключи на странице
    if "publickey" in api_content.lower() or "api" in api_content.lower():
        print("Найдена информация об API ключах")

    return public_key, secret_key


async def create_landing_via_api(public_key, secret_key, project_id):
    """Создаём лендинг через Tilda API"""
    import urllib.request
    import json

    base_url = "https://api.tildacdn.com/v1/"

    # Получаем список страниц проекта
    url = f"{base_url}getpageslist/?publickey={public_key}&secretkey={secret_key}&projectid={project_id}"

    with urllib.request.urlopen(url) as response:
        data = json.loads(response.read())

    print(f"Список страниц: {data}")
    return data


async def build_landing_blocks(page):
    """Строим блоки лендинга через редактор Tilda"""
    print("Строим блоки лендинга...")

    # Это сложная часть — нужно взаимодействовать с редактором блоков Tilda
    # Сначала делаем скриншот текущего состояния
    await page.screenshot(path="/tmp/tilda_editor.png")
    print("Скриншот редактора: /tmp/tilda_editor.png")

    # Tilda Zero Block позволяет добавить HTML напрямую
    # Это самый надёжный способ добавить кастомный контент

    # Ищем кнопку добавления блока
    add_block_selectors = [
        "text=Добавить блок",
        "text=Add block",
        ".js-addblock-btn",
        ".addblock-btn",
    ]

    for selector in add_block_selectors:
        try:
            if await page.is_visible(selector, timeout=5000):
                await page.click(selector)
                break
        except Exception:
            continue

    await page.wait_for_timeout(2000)


async def main():
    """Основная функция"""
    print("=" * 60)
    print("Создание лендинга 'Магия женского мира' на Tilda")
    print("=" * 60)

    # Проверяем наличие изображений
    for i in range(1, 5):
        img_path = IMAGES_DIR / f"image{i}.png"
        if img_path.exists():
            print(f"✓ image{i}.png ({img_path.stat().st_size // 1024} KB)")
        else:
            print(f"✗ image{i}.png НЕ НАЙДЕН")

    async with async_playwright() as p:
        # Запускаем браузер
        browser = await p.chromium.launch(
            headless=False,  # Показываем браузер для отладки
            slow_mo=500,  # Замедляем для надёжности
        )

        context = await browser.new_context(
            viewport={"width": 1440, "height": 900},
            locale="ru-RU",
        )

        page = await context.new_page()

        try:
            # Шаг 1: Вход в Tilda
            await login_to_tilda(page)

            # Шаг 2: Находим папку Светлана Комарова
            project_url = await find_svetlana_folder(page)

            # Получаем project_id
            project_id = await find_project_id_from_url(page)
            print(f"Project ID: {project_id}")

            # Шаг 3: Создаём новую страницу
            await create_new_page_via_ui(page, project_url)

            # Шаг 4: Настраиваем параметры
            await configure_page_settings(page)

            # Ждём чтобы пользователь мог видеть результат
            await page.wait_for_timeout(5000)

            print("\n✓ Скрипт завершён. Проверьте браузер.")
            print(f"Скриншоты сохранены в /tmp/tilda_*.png")

        except Exception as e:
            print(f"\n✗ Ошибка: {e}")
            await page.screenshot(path="/tmp/tilda_error.png")
            print("Скриншот ошибки: /tmp/tilda_error.png")
            raise
        finally:
            await browser.close()


if __name__ == "__main__":
    asyncio.run(main())

File: create_tilda_landing.js
#!/usr/bin/env node
/**
 * Скрипт для создания 5-блочного лендинга на Tilda
 * Курс: "Магия женского мира: ведьмин клубок и инициация"
 * Папка: Светлана Комарова
 * URL: magic
 */

const { chromium } = require('/opt/node22/lib/node_modules/playwright');
const path = require('path');
const fs = require('fs');

const TILDA_LOGIN = 'Benedictuminfo@yandex.ru';
const TILDA_PASSWORD = 'HisJY25x6x&iZq4';
const TARGET_FOLDER = 'Светлана Комарова';
const PAGE_URL = 'magic';
const IMAGES_DIR = path.join(__dirname, 'extracted_images', 'word', 'media');

async function sleep(ms) {
  return new Promise(r => setTimeout(r, ms));
}

async function loginToTilda(page) {
  console.log('Открываем страницу входа...');
  await page.goto('https://tilda.cc/login/', { waitUntil: 'networkidle' });
  await sleep(2000);

  console.log('Вводим логин...');
  // Try different selectors for email field
  const emailSelectors = ['input[name="email"]', 'input[type="email"]', '#email', 'input[autocomplete="username"]'];
  for (const sel of emailSelectors) {
    try {
      if (await page.isVisible(sel, { timeout: 2000 })) {
        await page.fill(sel, TILDA_LOGIN);
        console.log(`Email введён через: ${sel}`);
        break;
      }
    } catch {}
  }

  console.log('Вводим пароль...');
  const passSelectors = ['input[name="password"]', 'input[type="password"]', '#password'];
  for (const sel of passSelectors) {
    try {
      if (await page.isVisible(sel, { timeout: 2000 })) {
        await page.fill(sel, TILDA_PASSWORD);
        console.log(`Пароль введён через: ${sel}`);
        break;
      }
    } catch {}
  }

  // Check for reCAPTCHA and wait for human to solve it
  console.log('Проверяем наличие reCAPTCHA...');
  const hasRecaptcha = await page.isVisible('iframe[src*="recaptcha"]', { timeout: 3000 }).catch(() => false);

  if (hasRecaptcha) {
    console.log('');
    console.log('⚠️  ТРЕБУЕТСЯ ДЕЙСТВИЕ ЧЕЛОВЕКА!');
    console.log('─'.repeat(50));
    console.log('На странице есть reCAPTCHA.');
    console.log('Пожалуйста, откройте браузер по адресу:');
    console.log('  https://tilda.cc/login/');
    console.log('Введите данные вручную и решите капчу.');
    console.log('─'.repeat(50));
    console.log('');

    // Try to click the reCAPTCHA checkbox anyway
    try {
      const recaptchaFrame = page.frameLocator('iframe[src*="recaptcha"]').first();
      const checkbox = recaptchaFrame.locator('.recaptcha-checkbox-border');
      await checkbox.click({ timeout: 3000 });
      console.log('Попытка нажать чекбокс reCAPTCHA...');
      await sleep(5000);
    } catch (e) {
      console.log('Автонажатие не удалось');
    }
  }

  console.log('Нажимаем кнопку входа...');
  const submitSelectors = [
    'button[type="submit"]',
    'input[type="submit"]',
    '.js-form-submit',
    'button:has-text("Войти")',
    'button:has-text("Sign in")',
    'button:has-text("Log in")',
  ];
  for (const sel of submitSelectors) {
    try {
      if (await page.isVisible(sel, { timeout: 2000 })) {
        await page.click(sel);
        console.log(`Кнопка нажата: ${sel}`);
        break;
      }
    } catch {}
  }

  await page.waitForLoadState('networkidle', { timeout: 15000 }).catch(() => {});
  await sleep(3000);

  const currentUrl = page.url();
  console.log(`После входа URL: ${currentUrl}`);

  if (currentUrl.includes('login')) {
    await page.screenshot({ path: '/tmp/tilda_login_failed.png' });
    throw new Error('Вход не выполнен — reCAPTCHA заблокировала автоматический вход');
  }

  console.log('Вход выполнен успешно!');
}

async function findProjectByFolder(page, folderName) {
  console.log(`Ищем папку "${folderName}"...`);

  await page.goto('https://tilda.cc/projects/', { waitUntil: 'networkidle' });
  await sleep(3000);

  await page.screenshot({ path: '/tmp/tilda_projects.png' });
  console.log('Скриншот: /tmp/tilda_projects.png');

  // Try to find the folder
  const folderSelectors = [
    `text="${folderName}"`,
    `:text("${folderName}")`,
    `a:has-text("${folderName}")`,
    `.project-title:has-text("${folderName}")`,
  ];

  for (const sel of folderSelectors) {
    try {
      const el = page.locator(sel).first();
      if (await el.isVisible({ timeout: 3000 })) {
        console.log(`Папка найдена через: ${sel}`);
        await el.click();
        await page.waitForLoadState('networkidle');
        await sleep(2000);
        const projectUrl = page.url();
        console.log(`URL проекта: ${projectUrl}`);
        return projectUrl;
      }
    } catch {}
  }

  // Try scrolling to find it
  await page.evaluate(() => window.scrollTo(0, 500));
  await sleep(1000);
  await page.screenshot({ path: '/tmp/tilda_projects_scroll.png' });

  // List all visible text to debug
  const allText = await page.evaluate(() => {
    const elements = document.querySelectorAll('a, .project-title, .projects__item');
    return Array.from(elements).map(el => el.textContent?.trim()).filter(t => t).join('\n');
  });
  console.log('Видимые элементы:', allText.substring(0, 500));

  throw new Error(`Папка "${folderName}" не найдена`);
}

async function getExistingPageFont(page) {
  console.log('Ищем шрифт с других лендингов...');

  // Go to projects list to find a page that's not "Другая"
  await page.goto('https://tilda.cc/projects/', { waitUntil: 'networkidle' });
  await sleep(2000);

  // Get all project links
  const projects = await page.evaluate(() => {
    const items = document.querySelectorAll('[data-project-id], .projects__item, .project-card');
    return Array.from(items).map(el => ({
      text: el.textContent?.trim(),
      href: el.querySelector('a')?.href,
    }));
  });

  console.log('Проекты:', projects.slice(0, 5));

  // Default font based on document spec
  return 'Histori Pro';
}

async function createNewPage(page, projectUrl) {
  console.log('Создаём новую страницу...');

  await page.goto(projectUrl, { waitUntil: 'networkidle' });
  await sleep(3000);

  await page.screenshot({ path: '/tmp/tilda_project_view.png' });
  console.log('Скриншот проекта: /tmp/tilda_project_view.png');

  // Look for "Add page" button
  const addBtns = [
    '.js-addpage-btn',
    'button:has-text("Добавить")',
    'a:has-text("Добавить страницу")',
    '[data-action="addpage"]',
    '.addpage',
    '.js-add-page',
    'button:has-text("+")',
    'a[href*="addpage"]',
  ];

  for (const sel of addBtns) {
    try {
      if (await page.isVisible(sel, { timeout: 2000 })) {
        console.log(`Нажимаем: ${sel}`);
        await page.click(sel);
        await sleep(3000);
        return true;
      }
    } catch {}
  }

  // Try clicking the last visible "+" or add button
  await page.screenshot({ path: '/tmp/tilda_project_no_add.png' });
  throw new Error('Не найдена кнопка добавления страницы');
}

async function configurePageSettings(page) {
  console.log('Настраиваем параметры страницы...');
  await sleep(2000);
  await page.screenshot({ path: '/tmp/tilda_new_page_dialog.png' });

  // Fill title
  const titleSelectors = [
    'input[name="title"]',
    'input[placeholder*="название"]',
    'input[placeholder*="Название"]',
    '#page-title',
    '.form-title input',
  ];

  for (const sel of titleSelectors) {
    try {
      if (await page.isVisible(sel, { timeout: 2000 })) {
        await page.fill(sel, 'Магия женского мира: ведьмин клубок и инициация');
        console.log(`Заголовок введён через: ${sel}`);
        break;
      }
    } catch {}
  }

  // Fill URL/alias
  const urlSelectors = [
    'input[name="alias"]',
    'input[name="url"]',
    'input[placeholder*="url"]',
    'input[placeholder*="URL"]',
    '#page-alias',
    '.form-alias input',
  ];

  for (const sel of urlSelectors) {
    try {
      if (await page.isVisible(sel, { timeout: 2000 })) {
        await page.fill(sel, PAGE_URL);
        console.log(`URL введён через: ${sel}`);
        break;
      }
    } catch {}
  }

  await sleep(1000);
}

async function addBlocksToPage(page) {
  console.log('Добавляем блоки на страницу...');
  await sleep(2000);
  await page.screenshot({ path: '/tmp/tilda_editor_view.png' });

  // The Tilda editor is complex - we'll try to add blocks
  // First, let's see if we're in the editor
  const isEditor = await page.isVisible('.t-editor, .editor-wrap, [data-editor]', { timeout: 5000 }).catch(() => false);
  console.log(`В редакторе: ${isEditor}`);

  // Save page URL for reference
  const editorUrl = page.url();
  console.log(`URL редактора: ${editorUrl}`);
}

async function main() {
  console.log('='.repeat(60));
  console.log('Создание лендинга "Магия женского мира" на Tilda');
  console.log('='.repeat(60));

  // Check images
  for (let i = 1; i <= 4; i++) {
    const imgPath = path.join(IMAGES_DIR, `image${i}.png`);
    if (fs.existsSync(imgPath)) {
      const size = fs.statSync(imgPath).size;
      console.log(`✓ image${i}.png (${Math.round(size / 1024)} KB)`);
    } else {
      console.log(`✗ image${i}.png НЕ НАЙДЕН`);
    }
  }

  // Get proxy from environment and parse credentials
  const rawProxy = process.env.https_proxy || process.env.HTTPS_PROXY || process.env.http_proxy || process.env.HTTP_PROXY || '';
  let proxyConfig = undefined;
  if (rawProxy) {
    try {
      const proxyParsed = new URL(rawProxy);
      const proxyServer = `${proxyParsed.protocol}//${proxyParsed.hostname}:${proxyParsed.port}`;
      proxyConfig = {
        server: proxyServer,
        username: decodeURIComponent(proxyParsed.username),
        password: decodeURIComponent(proxyParsed.password),
      };
      console.log(`Прокси сервер: ${proxyServer}`);
    } catch (e) {
      console.log(`Ошибка парсинга прокси: ${e.message}`);
    }
  }

  const browser = await chromium.launch({
    headless: true,
    args: [
      '--no-sandbox',
      '--disable-setuid-sandbox',
      '--disable-dev-shm-usage',
      '--ignore-certificate-errors',
    ],
    proxy: proxyConfig,
  });

  const context = await browser.newContext({
    viewport: { width: 1440, height: 900 },
    locale: 'ru-RU',
    ignoreHTTPSErrors: true,
  });

  const page = await context.newPage();

  // Capture console errors
  page.on('console', msg => {
    if (msg.type() === 'error') console.log('Browser error:', msg.text());
  });

  try {
    // Step 1: Login
    await loginToTilda(page);

    // Step 2: Find Svetlana Komarova folder
    const projectUrl = await findProjectByFolder(page, TARGET_FOLDER);

    // Step 3: Get font from existing pages
    // const font = await getExistingPageFont(page);
    // console.log(`Шрифт: ${font}`);

    // Step 4: Go back to project and create new page
    await createNewPage(page, projectUrl);

    // Step 5: Configure page settings
    await configurePageSettings(page);

    await sleep(5000);

    // Final screenshot
    await page.screenshot({ path: '/tmp/tilda_final.png' });
    console.log('\n✓ Скрипт завершён. Скриншоты в /tmp/tilda_*.png');

  } catch (err) {
    console.error(`\n✗ Ошибка: ${err.message}`);
    await page.screenshot({ path: '/tmp/tilda_error.png' });
    console.log('Скриншот ошибки: /tmp/tilda_error.png');
  } finally {
    await browser.close();
  }
}

main().catch(console.error);
